crates/wasm-mutate/src/mutators/mod.rs

```diff
@@ -1,350 +1,425 @@
-                )*
-                _ => panic!("This mutator cannot be applied to this section")
+        match std::str::from_utf8(&str) {
+            Ok(s) => String::from(s),
+            Err(e) => {
+                let i = e.valid_up_to();
+                let valid = &str[0..i];
+                let s = unsafe {
+                    debug_assert!(std::str::from_utf8(valid).is_ok());
+                    std::str::from_utf8_unchecked(valid)
+                };
+                String::from(s)
             }
-            
-            consumed += chunksize;
+        }
+    }
+}
 
-            if consumed == sectionsize {
-                break
+impl Mutator for RenameExportMutator {
+    fn mutate<'a>(&self, config: &WasmMutate, info: &mut ModuleInfo<'a>) -> Module {
+        let mut exports = ExportSection::new();
+        let mut reader = ExportSectionReader::new(info.get_exports_section().data, 0).unwrap();
+        let max_exports = reader.get_count() as u64;
+        let skip_at = config.get_rnd().gen_range(0, max_exports);
+
+        (0..max_exports).for_each(|i| {
+            let export = reader.read().unwrap();
+
+            let new_name = if skip_at != i {
+                // otherwise bypass
+                String::from(export.field)
+            } else {
+                let new_name = self.limited_string(&mut config.get_rnd());
+                log::debug!("Renaming export {:?} by {:?}", export, new_name);
+                new_name
+            };
+
+            match export.kind {
+                wasmparser::ExternalKind::Function => {
+                    exports.export(new_name.as_str(), Export::Function(export.index));
+                }
+                wasmparser::ExternalKind::Table => {
+                    exports.export(new_name.as_str(), Export::Table(export.index));
+                }
+                wasmparser::ExternalKind::Memory => {
+                    exports.export(new_name.as_str(), Export::Memory(export.index));
+                }
+                wasmparser::ExternalKind::Global => {
+                    exports.export(new_name.as_str(), Export::Global(export.index));
+                }
+                wasmparser::ExternalKind::Module => {
+                    exports.export(new_name.as_str(), Export::Module(export.index));
+                }
+                wasmparser::ExternalKind::Instance => {
+                    exports.export(new_name.as_str(), Export::Instance(export.index));
+                }
+                _ => {
+                    panic!("Unknown export {:?}", export)
+                }
             }
-        }
-    };
+        });
+
+        // Ugly solution
+        let mut modu = Module::new();
+        info.raw_sections.iter().enumerate().for_each(|(idx, s)| {
+            // if section if this one...replace
+            if info.exports.unwrap() == idx {
+                modu.section(&exports);
+            } else {
+                modu.section(s);
+            }
+        });
```

Comment:
I think we can create a helper method on `ModuleInfo` to replace the `i`th section and return the resulting `wasm_encoder::Module`:

```rust
fn replace_section(&self, i: usize, new_section: &impl wasm_encoder::Section) -> wasm_encoder::Module {
    let mut module = wasm_encoder::Module::new();
    self.raw_sections.iter().enumerate().for_each(|j, s| {
        if i == j {
            module.section(replacement);
        } else {
            module.section(s);
        }
    });
    module
}
```

This should de-duplicate most of the end of these mutators impls so they can just be something like:

```rust
info.replace_section(info.exports.unwrap(), exports)
```